### PR TITLE
Enable using built-in libtiff on Linux

### PR DESCRIPTION
**Pull request by MaartenBent**
Required to make https://github.com/wxWidgets/wxWidgets/commit/f25a88a13dc51d9d857e10b4b264c0e817bb7a0f actually work

### CODE REVIEW

File: libtiff/tif_config.h [hunk 1]
```diff
@@ -3,7 +3,7 @@
     hardcoded definitions for MSVC, Xcode and MinGW when using makefiles.
  */
 
-#ifdef __APPLE__
+#if defined(__APPLE__) || defined(__linux__)
```
Comment (vadz) [inline libtiff/tif_config.h]: At this stage, should we test for `!defined(_WIN32)` instead perhaps?

**Review by vadz — COMMENTED** [review]
No comments, thanks!

File: libtiff/tif_config.h (hunk 1 above)
Comment (MaartenBent) [reply libtiff/tif_config.h]: Yes. I could also swap the cases around and test `#ifdef _WIN32`. But the diff will be bigger. Do you object?

Comment (vadz) [reply libtiff/tif_config.h]: No, I think we should avoid handling Linux specifically without any real need, other Unices still exist (albeit barely)...

**Pull request merged**